nyx_client/client.py

```diff
@@ -619,6 +620,13 @@ def get_data_for_creators(self, creators: list[str]) -> list[Data]:
         Returns:
             A list of `Data` instances matching the specified creators.
         """
+        if not creators:
+            return []
+        if len(creators) == 1:
+            # EI-3364 - a single entry results in the query not projecting the creator variable. An additional empty
+            # filter will prevent this whilst not matching anything (since the creator field cannot be empty).
+            creators += [""]
```

Comment:
does the same happen, for example, with `get_subscribed_data_for_creators`? 
and what if there is a single subscribed dataset in those filters?

Reply:
Good question. No, it only happens here - I tested all of the other `get_` methods. (It's to do with the order of statements - see our internal bug ticket.)

Reply:
It doesn't happen in the other one because it has more than one `FILTER` condition.